Sources/Filters/General/ClipClosedSurface/index.js

```diff
@@ -0,0 +1,1065 @@
+import macro from 'vtk.js/Sources/macros';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import { VtkDataTypes } from 'vtk.js/Sources/Common/Core/DataArray/Constants';
+import vtkDataArray from 'vtk.js/Sources/Common/Core/DataArray';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkContourTriangulator from 'vtk.js/Sources/Filters/General/ContourTriangulator';
+
+import Constants from './Constants';
+import CCSEdgeLocator from './ccsEdgeLocator';
+
+const { vtkErrorMacro } = macro;
+const { ScalarMode } = Constants;
+
+function vtkClipClosedSurface(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkClipClosedSurface');
+
+  // TODO: Check
+  publicAPI.getMTime = () =>
+    model.clippingPlanes.reduce(
+      (a, b) => (b.data.getMTime() > a ? b.data.getMTime() : a),
+      model.mtime
+    );
+
+  function squeezeOutputPoints(output, points, pointData, outputPointDataType) {
+    // Create a list of points used by cells
+    const n = points.getNumberOfPoints();
+    let numNewPoints = 0;
+
+    const outPointData = output.getPointData();
+    const pointMap = [];
+    pointMap.length = n;
+    pointMap.fill(-1);
+
+    const cellArrays = [
+      output.getVerts(),
+      output.getLines(),
+      output.getPolys(),
+      output.getStrips(),
+    ];
+    // Find all the newPoints that are used by cells
+    // TODO: Check
+    cellArrays.forEach((cellArray) => {
+      if (!cellArray) {
+        return;
+      }
+      const values = cellArray.getData();
+      let numPts;
+      for (let i = 0; i < values.length; i += numPts + 1) {
+        numPts = values[i];
+        for (let j = 1; j < numPts + 1; j++) {
+          const pointId = values[i + j];
+          if (pointMap[pointId] < 0) {
+            pointMap[pointId] = numNewPoints++;
+          }
+        }
+      }
+    });
+
+    // Create exactly the number of points that are required
+    const newPoints = vtkPoints.newInstance();
```

Comment:
`newInstance({size: numNewPoints})`

Reply:
Shouldn't it be `size: numNewPoints * 3`?

Reply:
indeed